dataprep/clean/clean_country.py

```diff
@@ -0,0 +1,328 @@
+    # pylint: disable=too-many-arguments
+    # check_country parses the value in row[col], and will return the index of the country
+    # in the DATA dataframe. The returned value "status" can be either "null"
+    # (which means row[col] contains a null value), "unknown" (in which case the value
+    # in row[col] could not be parsed) or "success" (a successful parse of the value).
+
+    country = str(row[col])
+    country = country.lower().strip()
+    result_index, status = check_country(country, input_format, True)
+
+    if fuzzy_dist > 0 and status == "unknown":
+        result_index, status = check_fuzzy_dist(country, input_format, fuzzy_dist)
+
+    if status == "null":
+        STATS["null"] += 1
+        row[f"{col}_clean"] = np.nan
+        return row
+    if status == "unknown":
+        if errors == "raise":
+            raise ValueError(f"unable to parse value {row[col]}")
+        STATS["unknown"] += 1
+        row[f"{col}_clean"] = row[col] if errors == "ignore" else np.nan
+        return row
+
+    result = DATA.iloc[result_index][output_format]
+    row[f"{col}_clean"] = str(int(result)) if output_format == "numeric" else result
+    if row[col] != row[f"{col}_clean"]:
+        STATS["cleaned"] += 1
+    return row
+
+
+@lru_cache(maxsize=2 ** 20)
+def check_country(country: str, input_format: str, clean: bool, strict: bool = False) -> Any:
+    """
+    Finds the index of the given country in the DATA dataframe.
+
+    Parameters
+    ----------
+    country
+        pandas Series of countries or str/int country value
+    input_format
+        the ISO 3166 input format of the country
+    clean
+        If True the function returns True/False to be used by the validate country function.
+        If False, a tuple (index, status) is returned.
+    strict
+        If True, the regex match for input types "name" and "offical" must occur at the
+        beginning of the input. If False, a match can be anywhere within the input.
+    """
+    if country in NULL_VALUES:
+        result = ("", "null") if clean else False
+        return result
+
+    if input_format == "auto":
+        input_format = get_format_from_name(country)
+
+    if input_format in ("regex", "name", "official"):
+        for index, regex in enumerate(REGEXES):
+            if not strict and regex.search(country):
+                result = (index, "success") if clean else True
+                return result
+            if strict and regex.match(country):
+                result = (index, "success") if clean else True
+                return result
+    else:
+        # convert to lowercase and remove 0's after decimal in floats (numeric column)
+        _match_col = DATA[input_format].astype(str).str.replace("\\..*", "")
+        indices = DATA[
+            _match_col.str.contains("^" + country + "$", flags=re.IGNORECASE, na=False)
+        ].index
+
+        if np.size(indices) > 0:
+            return (indices[0], "success") if clean else True
```

Comment:
Here `(indices[0], "success") if clean else True` is immediately returned, but elsewhere in this function, eg line 215, the returned value is stored in the variable `result`, and then `result` is returned. I suggest making this consistent, and I prefer immediately returning since I think storing the result in a variable is unnecessary.